Add model with cli tool

jlamypoirier opened the pull request.

jlamypoirier committed:
Add model with cli tool

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_1). All of your documentation changes will be reflected on that endpoint.

jlamypoirier committed:
Remove unwanted stuff

jlamypoirier committed:
Add new code

jlamypoirier committed:
Remove inference runner

jlamypoirier committed:
Style

Rocketknight1 committed:
Fix inverted conditional in TF common test! (#22540)

* Fix inverted conditional in TF common test!

* Make the same change in the PT tests file

* Make sure hidden states for GPT2 have the same output shape in PT/TF

* Minor fix to PT implementation of token classification loss

* Skip loss equivalence test for TFHubert because it keeps overflowing to inf

* Compute LM loss for TF the (weird) way it's computed in PT

* Skip loss equivalence test for Wav2Vec2 for the same reason as Hubert

* Fix - don't try to access the hidden states property when output is a tuple

jlamypoirier committed:
Fix checks

jlamypoirier committed:
Test updates

sgugger committed:
Skip failing test

gante committed:
Generate: `TextIteratorStreamer` timeout (#22576)

younesbelkada committed:
make fixup

younesbelkada committed:
fix docs

younesbelkada committed:
fix doc

younesbelkada committed:
fix test

younesbelkada committed:
Merge remote-tracking branch 'upstream/main' into HEAD

younesbelkada committed:
hopefully fix pipeline tests

younesbelkada committed:
refactor

younesbelkada committed:
fix CIs

younesbelkada committed:
add comment

younesbelkada committed:
rename to `GPTBigCodeForCausalLM`

younesbelkada committed:
correct readme

younesbelkada committed:
make fixup + docs

younesbelkada committed:
make fixup

sgugger committed:
Move back doctest instructions to setup.cfg (#22587)

younesbelkada committed:
fixes

gante committed:
Tests: disable `accelerate_tests` mark warnings (#22585)

Rocketknight1 committed:
Fix PT-TF equivalence test for GPT1 (#22586)

* Re-enable skipped test and fix the hidden state shape issue

* Actually fix the bug instead of just doing something wrong

qmeeus committed:
Add thousands separator in training summary (#22583)

The logger prints a summary at the beginning of training that displays some info such as number of examples, number of parameters, total number of steps, etc. Those numbers can be quite large and difficult to read. I added a thousand separator to improve readability for the following:
- num_examples
- num_train_epochs
- per_device_train_batch_size
- total_train_batch_size
- max_steps
- num_trainable_params

wonhyeongseo committed:
docs: ko: complete `_toctree.yml` (#22581)

Co-authored-by: gabrielwithappy <102908949+gabrielwithappy@users.noreply.github.com>

mpenagar committed:
Sync preprocesses before loading the processor at run_speech_recognition_ctc.py (#21926)

* Update run_speech_recognition_ctc.py

Make sure all processes wait until data is saved before loading the processor from the output_dit

* Make sure all processes wait until data is saved before loading the processor from the output_dit

* Update run_speech_recognition_ctc.py

* Update run_speech_recognition_seq2seq.py

Rocketknight1 committed:
Fix a typo in one of the BLIP pretrained checkpoint names (#22588)

Fixes a typo in one of the BLIP pretrained checkpoint names

younesbelkada committed:
fixes

Narsil committed:
Adding support for BPE merge creation from scores instead of ids. (#22582)

* Adding support for BPE merge creation from scores instead of ids.

* Revert warn -> raise.

* Update src/transformers/convert_slow_tokenizer.py

* Quality.

younesbelkada committed:
Add DePlot + MatCha on `transformers` (#22528)

* add deplot + matcha on `transformers`

* more docs

* correct path

* Update docs/source/en/model_doc/deplot.mdx

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* fix

* use auto processor

* Update docs/source/en/model_doc/matcha.mdx

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* make fixup

* Update docs/source/en/model_doc/deplot.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* add correct names

---------

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>
Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

Rocketknight1 committed:
Use native TF checkpoints for the BLIP TF tests (#22593)

* Use native TF checkpoints for the TF tests

* Remove unneeded exceptions

kausmeows committed:
feat(model parallelism): moving the labels to the same device as the logits for gpt2 and bart (#22591)

jlamypoirier committed:
Remove pruning

jlamypoirier committed:
Remove import

jlamypoirier committed:
Doc updates

jlamypoirier committed:
More pruning removal

jlamypoirier committed:
Combine copies

jlamypoirier committed:
Single MQA implementation, remove kv cache pre-allocation and padding

jlamypoirier committed:
Update doc

jlamypoirier committed:
Revert refactor to match gpt2 style

jlamypoirier committed:
Merge back key and value caches, fix some type hints

jlamypoirier committed:
Update doc

jlamypoirier committed:
Fix position ids pith padding (PR 21080)

jlamypoirier committed:
Add conversion script temporarily

jlamypoirier committed:
Update conversion script

jlamypoirier committed:
Remove checkpoint conversion

jlamypoirier committed:
New model

jlamypoirier committed:
Fix MQA test

jlamypoirier committed:
Fix copies

Narsil committed:
Adding Llama FastTokenizer support. (#22264)

* Adding Llama FastTokenizer support.

- Requires https://github.com/huggingface/tokenizers/pull/1183 version
- Only support byte_fallback for llama, raise otherwise (safety net).
- Lots of questions are special tokens

How to test:

```python

from transformers.convert_slow_tokenizer import convert_slow_tokenizer
from transformers import AutoTokenizer
from tokenizers import Tokenizer

tokenizer = AutoTokenizer.from_pretrained("huggingface/llama-7b")

if False:
    new_tokenizer = Tokenizer.from_file("tok.json")
else:
    new_tokenizer = convert_slow_tokenizer(tokenizer)
    new_tokenizer.save("tok.json")

strings = [
    "This is a test",
    "生活的真谛是",
    "生活的真谛是[MASK]。",
    # XXX: This one is problematic because of special tokens
    # "<s> Something something",
]

for string in strings:
    encoded = tokenizer(string)["input_ids"]
    encoded2 = new_tokenizer.encode(string).ids

    assert encoded == encoded2, f"{encoded} != {encoded2}"

    decoded = tokenizer.decode(encoded)
    decoded2 = new_tokenizer.decode(encoded2)

    assert decoded.strip() == decoded2, f"{repr(decoded)} != {repr(decoded2)}"
```

The converter + some test script.

The test script.

Tmp save.

Adding Fast tokenizer + tests.

Adding the tokenization tests.

Correct combination.

Small fix.

Fixing tests.

Fixing with latest update.

Rebased.

fix copies + normalized added tokens  + copies.

Adding doc.

TMP.

Doc + split files.

Doc.

Versions + try import.

Fix Camembert + warnings -> Error.

Fix by ArthurZucker.

Not a decorator.

* Fixing comments.

* Adding more to docstring.

* Doc rewriting.

younesbelkada committed:
try fix tests

younesbelkada committed:
Merge remote-tracking branch 'upstream/main' into HEAD

younesbelkada committed:
FIX TEST!!

younesbelkada committed:
remove  `DoubleHeadsModel`

younesbelkada committed:
add MQA tests

younesbelkada committed:
add slow tests

younesbelkada committed:
clean up

younesbelkada committed:
add CPU checker

younesbelkada committed:
final fixes

younesbelkada committed:
fixes

- fix GPU issue
- fixed slow tests
- skip disk offload

younesbelkada committed:
fix final issue

jlamypoirier committed:
Simplify and comment baddbmm fix

jlamypoirier committed:
Remove unnecessary code

jlamypoirier committed:
Transpose tweaks

jlamypoirier committed:
Use beta=1 on cpu, improve tests
